### PR TITLE
Gpunufft2.0

### PR DESCRIPTION
**Pull request by chaithyagr**

### CODE REVIEW

**Comment by paquiteau** [comment]
Merge this after merging benchmark-tweak #62

File: src/mrinufft/operators/interfaces/gpunufft.py [hunk 1]
```diff
@@ -9,8 +10,45 @@
...
+    -------
+    np.ndarray or None
+        the pinned sensitivity maps
+    """
+
+    if smaps is None:
+        return None
+    smaps_ = smaps.T.reshape(-1, smaps.shape[0])
+    cp.cuda.set_pinned_memory_allocator(_allocator)
+    pinned_smaps = cx.empty_pinned(smaps_.shape, dtype=np.complex64)
```
Comment (chaithyagr) [inline src/mrinufft/operators/interfaces/gpunufft.py:46]: This must be order 'F' just like others

Comment (chaithyagr) [reply src/mrinufft/operators/interfaces/gpunufft.py:46]:
```suggestion
    pinned_smaps = cx.empty_pinned(smaps_.shape, dtype=np.complex64, order='F')
```

**Comment by chaithyagr** [comment]
Can you fix the CI, reviewing this now

File: src/mrinufft/operators/interfaces/cufinufft.py [hunk 2]
```diff
@@ -523,7 +523,7 @@ def __adj_op(self, coeffs_d, image_d):
             ret = self.raw_op.type1(coeffs_d, image_d)
         return ret
 
-    def data_consistency(self, image_data, obs_data):
+    def get_grad(self, image_data, obs_data):
```
Comment (chaithyagr) [inline src/mrinufft/operators/interfaces/cufinufft.py:526]: DId we fixate finally on this name?

**Review by chaithyagr — COMMENTED** [review]
LGTM except one naming convention

**Comment by chaithyagr** [comment]
Sadly, I cant exactly "review" and mark green as I only opened this PR xD

File: src/mrinufft/operators/interfaces/cufinufft.py (hunk 2 above)
Comment (paquiteau) [reply src/mrinufft/operators/interfaces/cufinufft.py:526]: we decided on `data_consistency` indeed.

Comment (paquiteau) [reply src/mrinufft/operators/interfaces/cufinufft.py:526]: This is addressed in the other PR

**Pull request merged**